src/pages/analysis/data-analysis.tsx

```diff
@@ -0,0 +1,204 @@
+            ...prev,
+            summary: lastMessage?.result?.response[0].response,
+            newSessionId: lastMessage?.result?.session_id,
+          };
+        });
+      }
+    } catch (err: any) {
+      console.error(err);
+      errorToast(err.message, [], globalContext);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  // get llm summary
+  const getLLMSummary = async ({
+    sqlText,
+    user_prompt,
+  }: {
+    sqlText: string;
+    user_prompt: string;
+  }) => {
+    setLoading(true);
+    try {
+      const response = await httpPost(session, `warehouse/ask/`, {
+        sql: sqlText,
+        user_prompt,
+      });
+      console.log(response, 'taskid');
+      if (response?.detail) {
+        errorToast(response.detail, [], globalContext);
+        return { error: 'ERROR' };
+      }
+      if (!response?.request_uuid) {
+        errorToast('Something went wrong', [], globalContext);
+        return { error: 'ERROR' };
+      }
+
+      successToast(`Data analysis initiated successfully`, [], globalContext);
+      await delay(3000);
+      pollForTaskRun(response.request_uuid);
+    } catch (err: any) {
+      console.error(err);
+      setLoading(false);
+      errorToast(err.message, [], globalContext);
+    }
+  };
+
+  return (
+    <>
+      <Box
+        sx={{
+          p: '3rem 3rem',
+          width: '100%',
+          display: 'flex',
+          gap: '1rem',
+        }}
+      >
+        <Box
+          sx={{
+            ...customCss,
+            width: '42%',
+            flexDirection: 'column',
+          }}
+        >
+          {/* Top saved Session Option */}
+          <TopBar handleOpenSavedSession={handleOpenSavedSession} handleNewSession={handleNewSession} />
+
+          {/* SQL write Area */}
+          <SqlWrite
+            getLLMSummary={getLLMSummary}
+            prompt={prompt}
+            newSessionId={newSessionId}
+            oldSessionMetaInfo={oldSessionMetaInfo}
+            loading={loading}
+          />
+        </Box>
+
+        {/* Final Summary */}
+        <LLMSummary
+          llmSummary={summary}
+          newSessionId={newSessionId}
+          oldSessionMetaInfo={oldSessionMetaInfo}
+          prompt={prompt}
+        />
+
+        {/* Loader full screen */}
+        {loading && (
+          <>
+            <FullPageBackground>
+              <CircularProgress sx={{ color: '#FFFFFF' }} />
+              <Typography
+                sx={{ fontWeight: '600', fontSize: '20px', color: '#FFFFFF' }}
+              >
+                Prepping your data output...
+              </Typography>
+            </FullPageBackground>
+          </>
+        )}
+
+        {/* Saved Session Dailog */}
+        {openSavedSessionDialog && (
+          <SavedSession
+            open={openSavedSessionDialog}
+            onClose={handleCloseSavedSession}
+            handleEditSession={handleEditSession}
+          />
+        )}
+      </Box>
+    </>
+  );
+}
+
```

Comment:
why are we not using react hook form for state management, we have been using it everywhere in the app. 

Idea is for us to not worm so much behind state managemenet logic and let the hook form handle